app/pages/learn/01_tutorial/07_rich_client_apps/02_introduction-javafx-animation.md

```diff
@@ -0,0 +1,265 @@
+
+A KeyFrame can have a name which then can be used to identify this `KeyFrame` in an animation, for example for starting from this specific `KeyFrame` using `playFrom(String cuePoint)`.
+It is also possible to provide an `onFinished` implementation, which will be invoked when hitting this cue point.
+
+### KeyValue
+A [KeyValue](javafxdoc:KeyValue) establishes a mapping between a `WritableValue<T>` and a target value of type `T`. This is used to define the change of a value.
+An `Interpolator` can be additionally defined to set the rate of change for this value. The `KeyValue` class is immutable.
+
+### Example
+This example of `Timeline` creates a `Circle` which moves 200px in x direction over the duration of 5 seconds:
+
+![TimelineExample](/assets/images/javafx/animation/timeline-example.gif)
+```java
+Circle circle = new Circle(50, 150, 10, Color.GREEN);
+
+KeyValue x = new KeyValue(circle.translateXProperty(), 200);
+KeyFrame frame = new KeyFrame(Duration.seconds(5), x);
+Timeline timeline = new Timeline(frame);
+
+timeline.play();
+```
+
+## Interpolator
+The [Interpolator](javafxdoc:Interpolator) abstract class defines the rate of change at which values change over time, influencing the smoothness of animations.
+
+It provides several built-in implementations:
+- [Interpolator.DISCRETE](javafxdoc:Interpolator.DISCRETE): The `DISCRETE` interpolator creates a **sudden** transition between values without any intermediate steps.
+- [Interpolator.LINEAR](javafxdoc:Interpolator.LINEAR): The `LINEAR` interpolator produces a **constant** rate of change between values over time.
+- [Interpolator.EASE_IN](javafxdoc:Interpolator.EASE_IN): The `EASE_IN` interpolator starts the animation slowly and accelerates as it progresses.
+- [Interpolator.EASE_OUT](javafxdoc:Interpolator.EASE_OUT): The `EASE_OUT` interpolator starts quickly and slows down as it progresses.
+- [Interpolator.EASE_BOTH](javafxdoc:Interpolator.EASE_BOTH): The `EASE_BOTH` interpolator starts slowly, accelerates in the middle and slows down towards the end. It combines the characteristics of `EASE_IN` and `EASE_OUT`.
+
+Additionally, there are two static factory methods for [SPLINE](javafxdoc:Interpolator.SPLINE) and [TANGENT](javafxdoc:Interpolator.TANGENT) interpolation.
+
+[//]: # (ToDo: sync gifs)
+Here is a visualization of the Interpolator using the example from [Timeline](#example):
+### Discrete
+![Discrete](/assets/images/javafx/animation/interpolator/discrete-example.gif)
+### Linear
+![Linear](/assets/images/javafx/animation/interpolator/linear-example.gif)
+### Ease In
+![EaseIn](/assets/images/javafx/animation/interpolator/ease-in-example.gif)
+### Ease Out
+![EaseOut](/assets/images/javafx/animation/interpolator/ease-out-example.gif)
+### Ease Both
+![EaseBoth](/assets/images/javafx/animation/interpolator/ease-both-example.gif)
```

Comment:
We should try to get these into the enumeration above. Having a bunch of additional sections here seems weird.

Reply:
Maybe get rid of the enumeration and take the same approach as the transition examples?
Creating subsection and having explanation and example at the same place.